src/app/(proper_react)/redesign/(authenticated)/user/dashboard/DashboardTopBanner.tsx

```diff
@@ -24,222 +28,261 @@ export type BannerContent =
+        <p>
+          <a href="https://mozilla.org/products/monitor/how-it-works/">
+            {l10n.getString(
+              "dashboard-top-banner-monitor-protects-your-even-more-learn-more"
+            )}
+          </a>
+        </p>
+      </div>
+    );
+  }
 
-  return (
-    <>
-      <div className={styles.container}>
-        <div className={styles.explainerContentWrapper}>
-          {content && !isFixedTab && (
-            <div className={styles.explainerContent}>
-              <h3>{content.headline}</h3>
-              <p>{content.description}</p>
+  if (relevantGuidedStep.id === "ScanResult") {
+    if (
+      typeof props.stepDeterminationData.latestScanData?.results.length ===
+        "number" &&
+      props.stepDeterminationData.latestScanData.results.length > 0
+    ) {
+      // If the user has run a data broker scan, their data was found at a data
+      // broker, and they have not yet addressed that, encourage them to do that:
+      return (
+        <div className={styles.explainerContent}>
+          <h3>
+            {l10n.getString("dashboard-top-banner-protect-your-data-title")}
+          </h3>
+          <p>
+            {l10n.getString(
+              "dashboard-top-banner-protect-your-data-description",
               {
-                // TODO: Add unit test when changing this code:
-                /* c8 ignore next 4 */
-                content.cta ? (
-                  <span className={styles.cta}>{content.cta}</span>
-                ) : (
-                  ""
-                )
+                data_breach_total_num: props.bannerData.dataBreachTotalNum,
+                data_broker_total_num: props.bannerData.dataBrokerTotalNum,
               }
-              {content.learnMore ? <p>{content.learnMore}</p> : ""}
-            </div>
-          )}
-          {isFixedTab && (
-            <ProgressCard
-              resolvedByYou={dataBreachFixedNum}
-              autoRemoved={dataBrokerFixedNum}
-              totalNumExposures={totalExposures}
-            />
+            )}
+          </p>
+          <div className={styles.cta}>
+            <Button href={relevantGuidedStep.href} small variant="primary">
+              {l10n.getString("dashboard-top-banner-protect-your-data-cta")}
+            </Button>
+          </div>
+        </div>
+      );
+      // The V8 coverage reporter appears to get confused by the `ignore start`
+      // below, and think this closing bracket is ignored as well:
+      /* c8 ignore next 2 */
+    }
+
+    /* c8 ignore start */
+    // If the user has run a data broker scan, but their data was not found,
+    // congratulate them and encourage them to see their results.
+    // TODO: Check with PM - this state is currently never shown, because users
+    //       are directed to their data breaches if they have no unresolved data
+    //       brokers. Should we be keeping track of the user "acknowledging"
+    //       their scan results or something?
```

Comment:
Actually, I was typing out this question for PM, and then realised when this state should be shown. Update incoming.

Reply:
To deal with this, I created https://github.com/mozilla/blurts-server/pull/3423, and then noticed that there is a whole bunch of other errors. So I'll postpone this until that is merged, after which we can create tickets for all of those.